### ISSUE TITLE
MXNet on Mobile Device

### ISSUE THREAD

**Issue by PetroWu** [dmlc/mxnet#4714]
## Environment info
Operating System: MacOS, iOS

Compiler: gcc-6

Package used (Python/R/Scala/Julia): Python

MXNet version: 0.7, 0.8, 0.9.2

Python version and distribution: Python 2.7

## Error Message:
I tried to use MXNet on iOS but found that the 'amalgamation' folder is not suitable for current version. It is because the whole project structure has changed a lot and some files are removed. I tried to fix it but encountered endless problems such as:

```
mxnet_predict-all.cc:119:8: error: unknown type name 'DotExp'
inline DotExp<TA, TB, ltrans, rtrans, MSHADOW_SCALAR_>
       ^
mxnet_predict-all.cc:119:14: error: expected unqualified-id
inline DotExp<TA, TB, ltrans, rtrans, MSHADOW_SCALAR_>
             ^
mxnet_predict-all.cc:16091:7: error: no member named 'ParamFieldInfo' in
      namespace 'nnvm::dmlc'; did you mean '::dmlc::ParamFieldInfo'?
using dmlc::ParamFieldInfo;
      ^~~~~~~~~~~~~~~~~~~~
      ::dmlc::ParamFieldInfo
mxnet_predict-all.cc:14388:8: note: '::dmlc::ParamFieldInfo' declared here
struct ParamFieldInfo {
       ^
mxnet_predict-all.cc:16327:16: error: no template named 'Registry' in namespace
      'nnvm::dmlc'; did you mean '::dmlc::Registry'?
  friend class dmlc::Registry<Op>;
               ^~~~~~~~~~~~~~
               ::dmlc::Registry

```
I also tried to use CMake to make libmxnet.a but failed to build it on xcode on error:

`target specifies product type 'com.apple.product-type.tool', but there's no such product type for the 'iphoneos' platform`

Some new operations and parameters can not be parsed using the single .cc file on

https://github.com/pppoe/WhatsThis-iOS

So is there any solution? Could anyone update the amalgamation module, please!

**Comment by xlvector** [comment]
you can refer to this PR. it works although it does not pass CI https://github.com/dmlc/mxnet/pull/4667

**Comment by piiswrong** [comment]
This should be fixed for v0.9.3 and later
